libnavigation-ui/src/main/java/com/mapbox/navigation/ui/speed/SpeedLimitView.java

```diff
@@ -0,0 +1,213 @@
+package com.mapbox.navigation.ui.speed;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Color;
+import android.graphics.drawable.GradientDrawable;
+import android.graphics.drawable.LayerDrawable;
+import android.text.SpannableStringBuilder;
+import android.text.style.RelativeSizeSpan;
+import android.text.style.StyleSpan;
+import android.util.AttributeSet;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.lifecycle.LifecycleObserver;
+
+import com.mapbox.navigation.base.speed.model.SpeedLimit;
+import com.mapbox.navigation.base.speed.model.SpeedLimitSign;
+import com.mapbox.navigation.base.speed.model.SpeedLimitUnit;
+import com.mapbox.navigation.ui.R;
+
+import static android.graphics.Typeface.BOLD;
+import static android.text.Spanned.SPAN_EXCLUSIVE_EXCLUSIVE;
+
+public class SpeedLimitView extends ConstraintLayout implements LifecycleObserver {
+
+  private static final double KILO_MILES_FACTOR = 0.621371;
+  private static final float RADIUS = 10f;
+  private static final int VIENNA_MUTCD_BORDER_INSET = 4;
+  private static final int MUTCD_INNER_BACKGROUND_INSET = 7;
+  private static final int VIENNA_INNER_BACKGROUND_INSET = 16;
+  private static final int VIENNA_MUTCD_OUTER_BACKGROUND_INSET = 0;
+  private static final int MUTCD_BORDER_COLOR = Color.BLACK;
+  private static final int VIENNA_BORDER_COLOR = Color.RED;
+  private TextView speedLimitText;
+  private ImageView speedLimitView;
+  private ConstraintLayout speedView;
+  private int speedLimitBackgroundColor;
+
+  public SpeedLimitView(Context context) {
+    this(context, null);
+  }
+
+  public SpeedLimitView(Context context, @Nullable AttributeSet attrs) {
+    this(context, attrs, -1);
+  }
+
+  public SpeedLimitView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+    super(context, attrs, defStyleAttr);
+    inflate(getContext(), R.layout.layout_speed_limit, this);
+    initAttributes(attrs);
+  }
+
+  /**
+   * Invoke to update speed limit view to show appropriate speed limit text considering the sign and unit.
+   * @param speedLimit object storing speed limit data
+   */
+  public void setSpeedLimit(@NonNull SpeedLimit speedLimit) {
+    Integer speedLimitKmph = speedLimit.getSpeedKmph();
+    if (speedLimitKmph != null) {
+      SpeedLimitSign speedLimitSign = speedLimit.getSpeedLimitSign();
+      speedLimitView.setImageDrawable(getSpeedDrawable(speedLimitSign));
+      SpeedLimitUnit speedLimitUnit = speedLimit.getSpeedLimitUnit();
+      String speed = getSpeedLimit(speedLimitSign, speedLimitUnit, speedLimitKmph);
```

Comment:
How does this implementation account for an indefinite speed limit (like on the Autobahn in Germany)? The iOS implementation [inserts an infinity symbol](https://github.com/mapbox/mapbox-navigation-ios/blob/ae18aaa66a1b752dbbe62d4402a857b7a3a96f60/MapboxNavigation/SpeedLimitView.swift#L112-L113) in that case for expediency, although <img src="https://upload.wikimedia.org/wikipedia/commons/d/d3/Zeichen_282_-_Ende_sämtlicher_Streckenverbote%2C_StVO_1970.svg" width="32" alt="Ende sämtlicher Streckenverbote" align="top"> would be more appropriate.

Reply:
I am not sure what nav native sends in this particular case and do I need to perform any logical operations based on the value I receive.

Reply:
I think that’s the question: if we simulate a route on the Autobahn, does the speed limit view end up informing the driver that there’s no speed limit (different from an unknown speed limit)?